bika/lims/browser/js/coffee/bika.lims.site.coffee

```diff
@@ -0,0 +1,829 @@
+### Please use this command to compile this file into the parent `js` directory:
+    coffee --no-header -w -o ../ -c bika.lims.site.coffee
+###
+
+
+class window.SiteView
+
+  load: =>
+    console.debug "SiteView::load"
+
+    # load translations
+    jarn.i18n.loadCatalog 'bika'
+    @_ = window.jarn.i18n.MessageFactory('bika')
+
+    # initialze the loading spinner
+    @init_spinner()
+
+    # initialze the client add overlay
+    @init_client_add_overlay()
+
+    # initialze the client combogrid
+    @init_client_combogrid()
+
+    # initialze datepickers
+    @init_datepickers()
+
+    # initialze reference definition selection
+    @init_referencedefinition()
+
+    # initialze department filtering
+    @init_department_filtering()
+
+    # bind the event handler to the elements
+    @bind_eventhandler()
+
+    # allowed keys for numeric fields
+    @allowed_keys = [
+      8,    # backspace
+      9,    # tab
+      13,   # enter
+      35,   # end
+      36,   # home
+      37,   # left arrow
+      39,   # right arrow
+      46,   # delete - We don't support the del key in Opera because del == . == 46.
+      44,   # ,
+      60,   # <
+      62,   # >
+      45,   # -
+      69,   # E
+      101,  # e,
+      61    # =
+    ]
+
+    # The name of the department filter cookies
+    @department_filter_cookie = "filter_by_department_info"
+    @department_filter_disabled_cookie = "dep_filter_disabled"
+
+
+  ### INITIALIZERS ###
+
+  bind_eventhandler: =>
+    ###
+     * Binds callbacks on elements
+    ###
+    console.debug "SiteView::bind_eventhandler"
+
+    # Analysis service popup
+    $('.service_title span:not(.before)').on 'click', @on_analysis_service_title_click
+
+    # ReferenceSample selection changed
+    $("#ReferenceDefinition\\:list").on "change", @on_reference_definition_list_change
+
+    # Numeric field events
+    $('.numeric').on 'keypress', @on_numeric_field_keypress
+    $('.numeric').on 'paste', @on_numeric_field_paste
+
+    # AT field events
+    $("input[name*='\\:int\'], .ArchetypesIntegerWidget input").on "keyup", @on_at_integer_field_keyup
+    $("input[name*='\\:float\'], .ArchetypesDecimalWidget input").on "keyup", @on_at_float_field_keyup
+
+    # Autocomplete events
+    # XXX Where is this used?
+    $("input.autocomplete").on "keydown", @on_autocomplete_keydown
```

Comment:
I've done some quick-search in the code and haven't found anything conclusive... comes from the old days of `analysisrequest.add.js` maybe?

Reply:
Maybe when there is only one item for selection when a field is a reference field and the referencewidget (that selection list popup) is used?